lib/report.js

```diff
@@ -62,8 +62,23 @@ class Report {
       try {
         const path = resolve(this.resolve, v8ScriptCov.url)
         const script = v8toIstanbul(path, this.wrapperLength)
-        script.applyCoverage(v8ScriptCov.functions)
-        map.merge(script.toIstanbul())
+
+        if (v8ScriptCov.functions.every(({ functionName, ranges }) => {
+          if (!script.source.includes(functionName)) {
+            return false
+          }
+
+          for (const { endOffset } of ranges) {
+            if (endOffset > script.source.length) {
```

Comment:
Second check: If the file size is smaller than coverage's `endOffset`, that means the coverage is generated form the CJS-ESM wrapper.